pkg/collector/corechecks/oracle/activity.go

```diff
@@ -0,0 +1,213 @@
+	Comments       []string `json:"dd_comments,omitempty"`
+	QueryTruncated string   `json:"query_truncated,omitempty"`
+}
+
+type OracleActivityRow struct {
+	SessionID              uint64  `db:"SID" json:"sid,omitempty"`
+	SessionSerial          uint64  `db:"SERIAL#" json:"serial,omitempty"`
+	User                   *string `db:"USERNAME" json:"user,omitempty"`
+	Status                 string  `db:"STATUS" json:"status"`
+	OsUser                 *string `db:"OSUSER" json:"os_user,omitempty"`
+	Process                *string `db:"PROCESS" json:"process,omitempty"`
+	Client                 *string `db:"MACHINE" json:"client,omitempty"`
+	Port                   *string `db:"PORT" json:"port,omitempty"`
+	Program                *string `db:"PROGRAM" json:"program,omitempty"`
+	Type                   *string `db:"TYPE" json:"type,omitempty"`
+	SqlID                  *string `db:"SQL_ID" json:"sql_id,omitempty"`
+	ForceMatchingSignature *uint64 `db:"FORCE_MATCHING_SIGNATURE" json:"force_matching_signature,omitempty"`
+	SqlPlanHashValue       *uint64 `db:"SQL_PLAN_HASH_VALUE" json:"sql_plan_hash_value,omitempty"`
+	SqlExecStart           *string `db:"SQL_EXEC_START" json:"sql_exec_start,omitempty"`
+	Module                 *string `db:"MODULE" json:"module,omitempty"`
+	Action                 *string `db:"ACTION" json:"action,omitempty"`
+	ClientInfo             *string `db:"CLIENT_INFO" json:"client_info,omitempty"`
+	LogonTime              *string `db:"LOGON_TIME" json:"logon_time,omitempty"`
+	ClientIdentifier       *string `db:"CLIENT_IDENTIFIER" json:"client_identifier,omitempty"`
+	BlockingInstance       *uint64 `db:"BLOCKING_INSTANCE" json:"blocking_instance,omitempty"`
+	BlockingSession        *uint64 `db:"BLOCKING_SESSION" json:"blocking_session,omitempty"`
+	FinalBlockingInstance  *uint64 `db:"FINAL_BLOCKING_INSTANCE" json:"final_blocking_instance,omitempty"`
+	FinalBlockingSession   *uint64 `db:"FINAL_BLOCKING_SESSION" json:"final_blocking_session,omitempty"`
+	WaitEvent              *string `db:"EVENT" json:"wait_event,omitempty"`
+	WaitEventGroup         *string `db:"WAIT_CLASS" json:"wait_event_group,omitempty"`
+	Statement              *string `db:"SQL_TEXT" json:"statement,omitempty"`
+	PdbName                *string `db:"PDB_NAME" json:"pdb_name,omitempty"`
+	QuerySignature         string  `json:"query_signature,omitempty"`
+	RowMetadata
+}
+
+// Metadata contains the metadata fields common to all events processed
+type Metadata struct {
+	Timestamp      float64 `json:"timestamp,omitempty"`
+	Host           string  `json:"host,omitempty"`
+	Source         string  `json:"ddsource,omitempty"`
+	DBMType        string  `json:"dbm_type,omitempty"`
+	DDAgentVersion string  `json:"ddagentversion,omitempty"`
+}
+
+/*
+type MetricSender struct {
+	sender           aggregator.Sender
+	hostname         string
+	submittedMetrics int
+}
+*/
+
+func (c *Check) SampleSession() error {
+	start := time.Now()
+
+	sessionSamples := []OracleActivityRow{}
```

Comment:
It might seem a little silly but I think it would make sense to use a different struct for reading from the database than the one used to make the API call to our backend. I know it means copying values from the db struct to the API struct but if you take a moment to picture our ultimate end goal of having the API be shared across backend and the integration, it would make sense for the db structs to be different. 

That way, the db structs can have the db annotations, handle null values/pointers and the API struct can be tailored to the API needs (less pointers, more usage of zero values, json annotations).

Reply:
fixed in https://github.com/DataDog/datadog-agent/commit/33b33178b83de729daba7f4408e2921f0ab69770